src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+            eff->workf[EFF_SCALE] = scale / 1000.0f;
+            eff->work[EFF_ARGS] = args;
+            eff->work[EFF_UNKS1] = 0;
+            eff->workf[EFF_DIST] = dist;
+            eff->workf[EFF_ROLL] = Rand_ZeroFloat(2.0f * M_PI);
+            eff->alpha = 0;
+            eff->frame = (s16)Rand_ZeroFloat(1000.0f);
+            break;
+        }
+    }
+}
+
+void BossTw_AddShieldBlastEffect(GlobalContext* globalCtx, Vec3f* initalPos, Vec3f* initalSpeed, Vec3f* accel,
+                                 f32 scale, f32 arg5, s16 alpha, s16 args) {
+    s16 i;
+    BossTwEEffect* eff;
+
+    for (i = 0, eff = globalCtx->specialEffects; i < ARRAY_COUNT(sTWEffects); i++, eff++) {
+        if (eff->type == TWEFF_NONE) {
+            eff->type = TWEFF_SHLD_BLST;
+            eff->pos = *initalPos;
+            eff->curSpeed = *initalSpeed;
+            eff->accel = *accel;
+            eff->workf[EFF_SCALE] = scale / 1000.0f;
+            eff->workf[EFF_DIST] = arg5 / 1000.0f;
+            eff->work[EFF_ARGS] = args;
+            eff->work[EFF_UNKS1] = 0;
+            eff->alpha = alpha;
+            eff->frame = (s16)Rand_ZeroFloat(1000.0f);
+            break;
+        }
+    }
+}
+
+void BossTw_AddShieldDeflectEffect(GlobalContext* globalCtx, f32 arg1, s16 arg2) {
+    s16 i;
+    s16 j;
+    BossTwEEffect* eff;
+    Player* player = PLAYER;
+
+    sShieldHitPos = player->bodyPartsPos[15];
+    sShieldHitYaw = player->actor.shape.rot.y;
+
+    for (i = 0; i < 8; i++) {
+        for (eff = globalCtx->specialEffects, j = 0; j < ARRAY_COUNT(sTWEffects); j++, eff++) {
+            if (eff->type == TWEFF_NONE) {
+                eff->type = TWEFF_SHLD_DEFL;
+                eff->pos = sShieldHitPos;
+                eff->curSpeed = sZeroVector;
+                eff->accel = sZeroVector;
+                eff->workf[EFF_ROLL] = i * 0.7853982f;
+                eff->workf[EFF_YAW] = 1.5707964f;
+                eff->workf[EFF_DIST] = 0.0f;
+                eff->workf[EFF_SCALE] = arg1 / 1000.0f;
+                eff->work[EFF_ARGS] = arg2;
+                eff->work[EFF_UNKS1] = 0;
+                eff->alpha = 255;
+                eff->frame = (s16)Rand_ZeroFloat(1000.0f);
+                break;
+            }
+        }
+    }
+}
+
+void BossTw_AddShieldHitEffect(GlobalContext* globalCtx, f32 arg1, s16 arg2) {
+    s16 i;
+    s16 j;
+    BossTwEEffect* eff;
+    Player* player = PLAYER;
+
+    sShieldHitPos = player->bodyPartsPos[15];
+    sShieldHitYaw = player->actor.shape.rot.y;
+
+    for (i = 0; i < 8; i++) {
+        for (eff = globalCtx->specialEffects, j = 0; j < ARRAY_COUNT(sTWEffects); j++, eff++) {
+            if (eff->type == TWEFF_NONE) {
+                eff->type = TWEFF_SHLD_HIT;
+                eff->pos = sShieldHitPos;
+                eff->curSpeed = sZeroVector;
+                eff->accel = sZeroVector;
+                eff->workf[EFF_ROLL] = i * (2.0f * M_PI / 8.0f);
+                eff->workf[EFF_YAW] = 1.5707964f;
```

Comment:
```suggestion
                eff->workf[EFF_ROLL] = i * (M_PI / 4.0f);
                eff->workf[EFF_YAW] = M_PI / 2.0f;
```
was there a reason to write it as 2/8?